adapters/silverpush/silverpush.go

```diff
@@ -0,0 +1,346 @@
+package silverpush
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	bidderConfig  = "sp_pb_ortb"
+	bidderVersion = "1.0.0"
+)
+
+type adapter struct {
+	bidderName string
+	endpoint   string
+}
+
+type SilverPushImpExt map[string]json.RawMessage
+
+type SilverPushReqExt struct {
+	PublisherId string  `json:"publisherId"`
+	BidFloor    float64 `json:"bidfloor"`
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{Message: "No impressions in bid request."}}
+	}
+
+	return a.ValidateAndProcessRequest(request)
+}
+
+// Validate and Process the request return []*adapters.RequestData and []error
+func (a *adapter) ValidateAndProcessRequest(request *openrtb2.BidRequest) ([]*adapters.RequestData, []error) {
+	imps := request.Imp
+	requests := make([]*adapters.RequestData, 0, len(imps))
+	errors := make([]error, 0, len(imps))
+
+	for _, imp := range imps {
+		impsByMediaType, err := impressionByMediaType(&imp)
+		if err != nil {
+			errors = append(errors, err)
```

Comment:
Do you want to exit early here for this kind of error?

Reply:
Yes ! ! Silverpush only support for video and Banner type mediaType. Error is only return when we are not receiving these types. 
`if imp.Banner == nil && imp.Video == nil {
		return nil, &errortypes.BadInput{Message: "Invalid MediaType. SilverPush only supports Banner, Video"}
	}`

Reply:
Sounds good. Please add a `continue` statement after appending the error to exclude these impressions from the request.

Reply:
Fixed
[commit](https://github.com/prebid/prebid-server/pull/2957/commits/786baef2e551736d40decb753127e6a90a580b8e)